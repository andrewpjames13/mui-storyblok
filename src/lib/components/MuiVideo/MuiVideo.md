MuiVideo example:

```js

  const url = 'https://www.youtube.com/watch?time_continue=2&v=g_HL8p2RbKY&feature=emb_logo';

  <MuiVideo
    url={url}
  />
```