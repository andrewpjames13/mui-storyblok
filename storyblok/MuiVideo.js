module.exports = {
  "name": "MuiVideo",
  "display_name": null,
  "updated_at": "2020-06-10T16:58:47.456Z",
  "schema": {
    "url": {
      "type": "text",
      "description": "Url for the video"
    },
    "height": {
      "type": "text",
      "description": "Height of the video",
      "default_value": "500px"
    },
    "width": {
      "type": "text",
      "description": "Width of the video.",
      "default_value": "500px"
    },
    "controls": {
      "type": "boolean",
      "description": "   * Set to true or false to display native player controls\n   * Controls cannot be hidden for Wistia videos",
      "display_name": ""
    },
    "playing": {
      "type": "boolean",
      "description": "* Set to true or false to pause or play the media."
    },
    "loop": {
      "type": "boolean",
      "description": "* Set to true or false to loop the media."
    },
    "light": {
      "type": "boolean",
      "description": "   * Set to true to show just the video thumbnail, which loads the full player on click\n   * Pass in an image URL to override the preview image"
    },
    "playbackRate": {
      "type": "number",
      "description": "   * Set the playback rate of the player\n   * Only supported by YouTube, Wistia, and file paths",
      "default_value": "1"
    },
    "playsinline": {
      "type": "boolean",
      "description": "* Applies the playsinline attribute where supported."
    },
    "pip": {
      "type": "boolean",
      "description": "   * Set to true or false to enable or disable picture-in-picture mode\n   * Only available when playing file URLs in certain browsers"
    },
    "rootClass": {
      "type": "options",
      "description": "* StoryBlok multiselect of CSS Classes\n* Applied to video Div"
    }
  },
  "image": null,
  "preview_field": null,
  "is_root": false,
  "preview_tmpl": null,
  "is_nestable": true,
  "all_presets": [],
  "preset_id": null,
  "real_name": "MuiVideo",
  "component_group_uuid": null
}