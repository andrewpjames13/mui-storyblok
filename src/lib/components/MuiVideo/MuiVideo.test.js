import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import MuiVideo from './MuiVideo';

function setup() {
  const props = {
    url: 'www.asdfiojasdf.com',
  };

  const comp = shallow(<MuiVideo {...props} />);
  return { comp, props };
}

describe('<MuiVideo />', () => {
  test('snapshot', () => {
    const { props } = setup();
    const tree = renderer.create(<MuiVideo {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it('should render MuiVideo', () => {
    const { comp } = setup();
    expect(comp).toMatchSnapshot();
  });
});
