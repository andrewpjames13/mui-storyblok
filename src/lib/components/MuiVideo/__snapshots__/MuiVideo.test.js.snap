// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MuiVideo /> should render MuiVideo 1`] = `ShallowWrapper {}`;

exports[`<MuiVideo /> snapshot 1`] = `
<div
  style={
    Object {
      "height": "500px",
      "width": "640px",
    }
  }
/>
`;
